### PR TITLE
usize -> u64 or u32

### PR DESCRIPTION
**Pull request by mimoo**
helps https://github.com/o1-labs/proof-systems/issues/346

### CODE REVIEW

File: kimchi/src/circuits/polynomials/chacha.rs [hunk 1]
```diff
@@ -554,7 +554,7 @@ mod tests {
                 h.insert(Column::Witness(i));
             }
             for i in 0..(lookup_info.max_per_row + 1) {
-                h.insert(Column::LookupSorted(i));
+                h.insert(Column::LookupSorted(i as usize));
```
Comment (mrmr1993) [inline kimchi/src/circuits/polynomials/chacha.rs:557]: Nit: `LookupSorted` should also become `u32` (although not necessarily in this PR)

File: kimchi/src/circuits/expr.rs [hunk 2]
```diff
@@ -169,7 +169,7 @@ pub enum ConstantExpr<F> {
     EndoCoefficient,
     Mds { row: usize, col: usize },
     Literal(F),
-    Pow(Box<ConstantExpr<F>>, usize),
+    Pow(Box<ConstantExpr<F>>, u64),
```
Comment (mrmr1993) [inline kimchi/src/circuits/expr.rs:172]: I don't think we ever use a large `Pow`, it might be safer to move to `u32`

**Review by mrmr1993 — APPROVED** [review]
Great, thank you for this!

File: kimchi/src/circuits/expr.rs (hunk 2 above)
Comment (mimoo) [reply kimchi/src/circuits/expr.rs:172]: I kept `u64` as this is what arkworks expect when we use `pow` there, but I don't see a problem moving this to `u32` either

File: kimchi/src/circuits/polynomials/chacha.rs (hunk 1 above)
Comment (mimoo) [reply kimchi/src/circuits/polynomials/chacha.rs:557]: I tried that but this led to another rabbit hole so made a minimal PR

**Pull request merged**